env/env_posix.cc

```diff
@@ -520,16 +520,17 @@ class PosixEnv : public Env {
       return Status::OK();
     }
 
-    switch (errno) {
+    int err = errno;
+    switch (err) {
       case EACCES:
       case ELOOP:
       case ENAMETOOLONG:
       case ENOENT:
       case ENOTDIR:
         return Status::NotFound();
       default:
-        assert(result == EIO || result == ENOMEM);
-        return Status::IOError("Unexpected error(" + ToString(result) +
+        assert(err == EIO || err == ENOMEM);
```

Comment:
Is `EROFS` possible?

Reply:
We should never see EROFS as we are just checking for the existence of a file/directory with F_OK. EFAULT and EINVAL are very much possible.